### PR TITLE
Update contract and tests for Ticket AttestedObject

### PR DESCRIPTION
**Pull request by JamesSmartCell**
Update for the SmartContract handling for the Ticket AttestedObject.
Tests are now passing; should also include a negative test.

Linked Issue: #207

### CODE REVIEW

**Review by jot2re — CHANGES_REQUESTED** [review]
It looks good! But as far as I can see only the ZKP is checked? 
I cannot seem to find the rest of the required checks done [here](https://github.com/TokenScript/attestation/blob/82cbe76d82d1df4a7c39a4873864a79264255d48/src/main/java/org/tokenscript/attestation/AttestedObject.java#L119) and [here](https://github.com/TokenScript/attestation/blob/82cbe76d82d1df4a7c39a4873864a79264255d48/src/main/java/org/tokenscript/attestation/AttestedObject.java#L153) being carried out.

**Comment by JamesSmartCell** [comment]
> It looks good! But as far as I can see only the ZKP is checked? I cannot seem to find the rest of the required checks done [here](https://github.com/TokenScript/attestation/blob/82cbe76d82d1df4a7c39a4873864a79264255d48/src/main/java/org/tokenscript/attestation/AttestedObject.java#L119) and [here](https://github.com/TokenScript/attestation/blob/82cbe76d82d1df4a7c39a4873864a79264255d48/src/main/java/org/tokenscript/attestation/AttestedObject.java#L153) being carried out.

These are now all tested. I have added a minting test with lots of negative tests to ensure all functionality doesn't get broken.

There is also a deployment script.

The tests are in a TypeScript hardhat harness; this may need to be added to the Travis CI.

**Comment by jot2re** [comment]
@JamesSmartCell I see the changes are thousands of lines so I think I need a pointer to the checking of the timestamp.
I would imagine there should be a function calling both the verification of the ZKP and timestamp and then returns the address of the issuer, attestor and subject where the issuer and attestor address then gets verified by the calling contract.

File: ethereum/attestation-test/hardhat.config.ts [hunk 1]
```diff
@@ -81,7 +81,10 @@ export default {
       accounts: [`${PRIVATE_KEY}`]
     },
     mumbai: {
-      url: `https://matic-mumbai.chainstacklabs.com`, //ths RPC seems to work more consistently
+      url: `https://rpc-mumbai.maticvigil.com`, //ths RPC seems to work more consistently
```
Comment (hboon) [inline ethereum/attestation-test/hardhat.config.ts:84]: There's also Infura:  `https://polygon-mumbai.infura.io/v3/#{key}`

**Review by oleggrib — APPROVED** [review]
@JamesSmartCell , it works good and we already use it. thanks

**Comment by JamesSmartCell** [comment]
@jot2re since the code is in use now, safe to merge?

**Comment by jot2re** [comment]
> @jot2re since the code is in use now, safe to merge?
@JamesSmartCell I would still like a pointer to the checks fo ZKP and timestamp, since it is really hard to find due to the amount of line changes.
Any pointers to the flow would be great. However, I can also verify this as part of the upcoming security review. But any kind of prose description of the specific smart contracts, their responsibility and interaction would greatly help me in this task and be much appreciated 😅

**Pull request merged**